src/cljs/main/broadfcui/common.cljs

```diff
@@ -210,3 +210,57 @@
                 :alert)
     [:span {:style {:fontWeight "bold" :fontSize "98%"}}
       "FireCloud is not intended to host personally identifiable information. Do not use any patient identifier, including name, social security number, or medical record number."]])
+
+(react/defc FoundationTooltip
+  {:component-did-mount
+   (fn [{:keys [this]}]
+     (.foundation (js/$ (react/find-dom-node this))))
+   :render
+   (fn [{:keys [props]}]
+     (let [{:keys [position text tooltip]} props]
+       [:span {:data-tooltip "" :className (str "has-tip " position)
+               :title tooltip}
+        text]))})
+
+(react/defc FoundationInfoBox
+  {:component-will-mount
+   (fn [{:keys [locals]}]
+     (swap! locals assoc :infobox-id (gensym "infobox-")))
+   :component-did-mount
+   (fn [{:keys [locals props]}]
+     (let [infobox-container (.createElement js/document "div")
+           infobox-id (:infobox-id @locals)]
+       (swap! locals assoc :infobox-container infobox-container)
+       (.insertBefore js/document.body infobox-container (.getElementById js/document "app"))
+       (react/render
+        (react/create-element [:div {:className "dropdown-pane" :id infobox-id :data-dropdown ""
+                                     :style {:whiteSpace "normal"}}
+                               (:text props)])
+        infobox-container
+        #(let [infobox-element (js/$ (str "#" infobox-id))
```

Comment:
Instead of using an ID for this, use the callback ref pattern you just established. I'm okay if you want to come back to this after your other work, but I don't want to finish the week without a PR for it.